plugins/rhs_plugin/index.js

```diff
@@ -12,13 +12,14 @@ function mapStateToProps(state) {
     const rhsPlugins = state.plugins.components.RightHandSidebarComponent;
     const pluginId = getPluginId(state);
 
-    const plugin = rhsPlugins.find((element) => element.id === pluginId);
-    const pluginName = plugin ? plugin.title : '';
-    const icons = plugin ? plugin.icons : [];
+    const pluginComponent = rhsPlugins.find((element) => element.pluginId === pluginId);
+    const pluggableId = pluginComponent ? pluginComponent.id : null;
```

Comment:
I might be misunderstanding this since they're both named the same, but does the change above (`(this.props.pluggableId !== undefined)`) conflict with this?

Reply:
We're passing in a `null` value here when we can't locate the respective plugin component in the store. `Pluggable` component attempts to find from its list of components if `pluggableId` field is set, or displays all the components (in a row) if it's not set (`undefined`). In order to prevent showing other components in the RHS while the webapp loads the new manifest, we simply set the value of `null` to indicate RHS to not load any component (dictated by the logic defined in `Pluggable`).

Unfortunately there is a subtle distinction between how JS defines `null` vs `undefined.

Treating `null` as setting a unknown/uninitialized value.
Treating `undefined` as the field was never set.

https://codeburst.io/javascript-null-vs-undefined-20f955215a2

Reply:
What I meant was, prior to the change, pluggable was testing `plugableId`. Now it's equality testing on undefined, so null passes through. 
I didn't read down to see plugableId only being used for an equality check, so nothing will break. So nevermind.

Reply:
ahh gotcha. Correct, null will be passed through now.

Reply:
Is there an opportunity to make this behaviour explicit? I find the juggling of `undefined` and `null` resulting in different outcomes to be relatively unexpected.

Reply:
Passing in an explicit `showPluggable` flag.